### ISSUE TITLE
string.Replace should not trigger the MA0074 rule

### ISSUE THREAD

**Issue by louis-z** [meziantou/Meziantou.Analyzer#555]
### Version of the Meziantou.Analyzer NuGet package

2.0.61

### Rule Identifier

MA0074

### Target Framework

net472

### C# Language version

C# 10

### Description

A MA0074 rule violation is reported when using `string.Replace`. According to [the doc](https://learn.microsoft.com/en-us/dotnet/api/system.string.replace?view=netframework-4.7.2#system-string-replace(system-string-system-string)), however, that method

> _[...]_ performs an ordinal (case-sensitive and culture-insensitive) search to find `oldValue`.

### Reproduction Steps

Minimal code:

```c#
var str1 = "foo";
var str2 = str1.Replace("foo", "bar");
```
